Utilities/bootstrap

```diff
@@ -193,8 +193,8 @@ def parse_build_args(args):
     if args.foundation_build_dir:
         args.foundation_build_dir = os.path.abspath(args.foundation_build_dir)
 
-    if args.llbuild_build_dir:
-        args.llbuild_build_dir = os.path.abspath(args.llbuild_build_dir)
+    if args.build_dirs["llbuild"]:
```

Comment:
This doesn't make sense: keep this one and the second one on the next line as it was, as you haven't initialized this before.

Reply:
Thanks for pointing this out! I haven't used Python for many years, so I'm quite rusty.

It turned out that `source_dirs` was not initialiized before use as well. Also, reverting `build_dirs["llbuild"]` back to `llbuild_build_dir` would change 13 locations in total, and would be inconsistent with other "build_dir" namings. So, instead, I initialized `source_dirs` and `build_dirs` as empty dictionaries in `parse_global_args`, and changed

```python
if args.build_dirs["llbuild"]
# and
if not args.build_dirs["llbuild"]
```

to 

```python
if "llbuild" in args.build_dirs
# and
if not "llbuild" in args.build_dirs
```

Do you think this change is ok?

Reply:
Not sure about Python initialization and dictionary checking, you'd have to look that up in the online docs and maybe the REPL.

My point about this variable is different: [`args.llbuild_build_dir` is already declared by the argument parser above](https://github.com/apple/swift-package-manager/blob/426e2963c63a3e2d65389f30dd63f7ba98976703/Utilities/bootstrap#L111). You are not setting `args.build_dirs["llbuild"]` from scratch like the `source_dirs` above, but checking to see if one was passed in, like `args.foundation_build_dir` in the line above.

So you can keep the 11 other locations using `args.build_dirs["llbuild"]` below, but you must first initialize it properly with `args.llbuild_build_dir` here, in case one was passed in, hence these two must be switched back. It will still build with the incorrect way you have it now, as `build_llbuild()` handles the case when a build directory wasn't passed in too, but you are ignoring a passed-in `args.llbuild_build_dir` the way you have it now.

Reply:
Thanks for this explanation, and sorry for misinterpreting your first comment! I changed the 2 `llbuild_build_dir`s back in cbbbff254d5f91775304218eeb7f8d696398b94c.